pkg/apis/controlplane/types.go

```diff
@@ -290,3 +290,29 @@ type NetworkPolicyNodeStatus struct {
 	// The generation realized by the Node.
 	Generation int64
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Group is the message format of antrea/pkg/controller/types.Group in an API response.
+type Group struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	// GroupMembers is a list of resources selected by this group.
+	GroupMembers []GroupMember
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// GroupPatch describes the incremental update of an Group.
+type GroupPatch struct {
```

Comment:
I think I'm missing something. I see several comments like this one: "internal Group types which can be later converted to AppliedToGroup or AddressGroup based on usage". So my assumption was that the Agent would never watch the `GroupPatch` API. Who's going to consume this API then?

Reply:
This should be consumed to report the realized members of a ClusterGroup and also for group associations of Pods.
@Dyanngg has a PR for that

Reply:
@Dyanngg I didn't realize we were planning to send incremental updates to group membership. I thought we would ask clients to poll groups and send the whole group every time. Has this changed?

Reply:
We should comment somewhere who consumes the Group/GroupPatch resources, and how they are consumed.

Reply:
My PR is not consuming this GroupPatch at the moment... My implementation is aligned with @antoninbas. I use a GroupMemberList struct to store members in the controlplane msg https://github.com/vmware-tanzu/antrea/pull/1688/commits/506a979ca56699b8d4c083ba0cde4185244b7ee5#diff-5540be2f630431d99bcd1d6cf91355ec0393231401e5964e28dd25b196be769d

Reply:
thanks will remove the GroupPatch and add comment shortly

Reply:
actually i will have to revisit the store usage here.. the Patch is necessitated by our store. will have to use cache.store for this case

Reply:
removed GroupPatch and instead return the latest copy of the object for modified events